src/Symfony/Component/Security/Http/EventListener/PasswordMigratingListener.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Symfony\Component\Security\Http\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\PasswordUpgradeBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\UserPassportInterface;
+use Symfony\Component\Security\Http\Event\LoginSuccessEvent;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @final
+ * @experimental in 5.1
+ */
+class PasswordMigratingListener implements EventSubscriberInterface
+{
+    private $encoderFactory;
+
+    public function __construct(EncoderFactoryInterface $encoderFactory)
+    {
+        $this->encoderFactory = $encoderFactory;
+    }
+
+    public function onLoginSuccess(LoginSuccessEvent $event): void
+    {
+        $passport = $event->getPassport();
+        if (!$passport instanceof UserPassportInterface || !$passport->hasBadge(PasswordUpgradeBadge::class)) {
+            return;
+        }
+
+        /** @var PasswordUpgradeBadge $badge */
+        $badge = $passport->getBadge(PasswordUpgradeBadge::class);
+        $plaintextPassword = $badge->getPlaintextPassword();
```

Comment:
why not directly erase the plain password when getting it? to avoid to forget to call it etc

Reply:
It seems a bit weird to have a getter with side-effects, but I think I'm 👍 on this idea

Reply:
I agree with Wouter. Such a method couldn't be called `getXXX()` anymore. Maybe `getAndErasePlaintextRemove()`, `getAndDeletePlaintextRemove()`, `getAndRemovePlaintextRemove()`, etc.?